### DISCUSSION TITLE
ofdm modulator function

### DISCUSSION THREAD

**Discussion by doogie05** [srsran/srsRAN_Project#774] [General Help]
I have a question on the implementation of OFDM modulator in srsRAN project source code.

Per OFDM architecture block diagram below,  for transmitter side, IFFT is performed on the modulated signals.
IFFT output is then fed into the DAC, upconverted and then transmitted.
![image](https://github.com/user-attachments/assets/0e62861b-3479-4947-866e-49af16cb983f)

ofdm modulator function below, performs DFT.  Looking at the DFT function, I was not sure DFT function performs Inverse DFT or DFT.
https://github.com/srsran/srsRAN_Project/blob/main/lib/phy/lower/modulation/ofdm_modulator_impl.cpp#L56

DFT function is here:
https://github.com/srsran/srsRAN_Project/blob/main/lib/phy/generic_functions/dft_processor_generic_impl.cpp#L31

I suspect that DFT() output is in time domain since cyclic prefix is copied at the end of the modulator function.  However, I need a confirmation on how Inverse DFT and DFT functions are implemented in srsRAN.

Thanks.
James

**Comment by rainey19** [comment]
At some point before that part of the phy is called, the fft library gets instantiated and configured (it uses FFTW by default). At that time, you configure the fft with parameters like size, buffer locations, and forward/inverse. Line 41 is where the DFT object gets brought in, you can trace the code calls backwards to see where it gets set up.

**Reply by doogie05** [reply]
Thanks for the suggestions.  What would be the easiest way to print out dft size?  e.g. https://github.com/srsran/srsRAN_Project/blob/main/lib/phy/lower/modulation/ofdm_modulator_impl.cpp#L68C3-L68C16
Is there anyway I can enable srsran_assert() without faulting out by updating docker configuration in docker compose file?

Thanks.
